thread-manager/src/native_thread_runtime.rs

```diff
@@ -0,0 +1,163 @@
+use {
+    crate::{
+        policy::{apply_policy, parse_policy, CoreAllocation},
+        MAX_THREAD_NAME_CHARS,
+    },
+    anyhow::bail,
+    log::warn,
+    serde::{Deserialize, Serialize},
+    solana_metrics::datapoint_info,
+    std::{
+        ops::Deref,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc, Mutex,
+        },
+    },
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct NativeConfig {
+    pub core_allocation: CoreAllocation,
+    pub max_threads: usize,
+    /// Priority in range 0..99
+    pub priority: u8,
+    pub policy: String,
+    pub stack_size_bytes: usize,
+}
+
+impl Default for NativeConfig {
+    fn default() -> Self {
+        Self {
+            core_allocation: CoreAllocation::OsDefault,
+            max_threads: 16,
+            priority: crate::policy::DEFAULT_PRIORITY,
+            policy: "OTHER".to_owned(),
+            stack_size_bytes: 2 * 1024 * 1024,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct NativeThreadRuntimeInner {
+    pub id_count: AtomicUsize,
+    pub running_count: Arc<AtomicUsize>,
+    pub config: NativeConfig,
+    pub name: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct NativeThreadRuntime {
+    inner: Arc<NativeThreadRuntimeInner>,
```

Comment:
afaict we never use the fact this an `Arc` (could be because no calling code yet)?

Can you explain the thought-process on needing an Arc-wrapped inner type here, vs just exposing `NativeThreadRuntimeInner` (we'd renamed as `NativeThreadRuntime`)

Reply:
same comment applies to the rayon runtime, but not tokio.

Reply:
This is indeed to simplify integration with agave (calling code which is coming in next PRs). The reason for this is to make integration into agave a bit less invasive. Most old code that spawns threads at will assumes ownership of thread pools, so the arc-wrapped type emulates this behavior to avoid adding lifetime annotations all over the place. It so happens that older code spawns native threads or rayon pools. Code that uses tokio mostly relies on runtime reference passed from caller, so it does not require such measures.